File: ChickInRun/Assets/BGM/bgmoper.js
#pragma strict
public var deathCheck : boolean;

function Start () {

}

function Update () {

}
function OnTriggerStay (col : Collider){
	if(deathCheck){
		//Debug.Log("Stop");
		GetComponent.<AudioSource>().Stop();
	}
}

function OnTriggerEnter (col : Collider){
	if(deathCheck){

	}
	else{
		if(col.gameObject.tag == "Chick"){
			//Debug.Log("Start");
    		GetComponent.<AudioSource>().Play();
		}
	}
}

function OnTriggerExit (col : Collider){
	if(col.gameObject.tag == "Chick"){
		//Debug.Log("Exit");
		GetComponent.<AudioSource>().Stop();
	}
}

File: ChickInRun/Assets/Script/BgmOper.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class BgmOper : MonoBehaviour
{
    public bool deathCheck;

    private void OnTriggerStay(Collider other)
    {
        if(deathCheck)
        {
            GetComponent<AudioSource>().Stop();
        }
    }

    private void OnTriggerEnter(Collider other)
    {
        if(!deathCheck)
        {
            if(other.gameObject.tag == "Chick")
            {
                GetComponent<AudioSource>().Play();
            }
        }
    }

    private void OnTriggerExit(Collider other)
    {
        if(other.gameObject.tag == "Chick")
        {
            GetComponent<AudioSource>().Stop();
        }
    }

    // Start is called before the first frame update
    void Start()
    {
        
    }

    // Update is called once per frame
    void Update()
    {
        
    }
}
